inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp

```diff
@@ -209,6 +210,34 @@ void LayerTestsCommon::Validate() {
     Compare(expectedOutputs, actualOutputs);
 }
 
+void LayerTestsCommon::checkKernel(std::string layerType, std::string permissibleKernel) {
+    std::vector<std::string> opNames;
+    auto perfomanceCounters = inferRequest.GetPerformanceCounts();
+    for (auto& pc : perfomanceCounters) {
+        if (pc.second.layer_type == layerType) {
+            opNames.push_back(pc.first);
+        }
+    }
+    ASSERT_TRUE(!opNames.empty())
+        << "layer with type " << layerType << " does not exist";
+
+    const std::regex expectedKernel(permissibleKernel);
+    for (auto& targetOperationName : opNames) {
+        auto targetOperationKernel = getKernel(targetOperationName);
+        EXPECT_TRUE(std::regex_search(targetOperationKernel.begin(), targetOperationKernel.end(), expectedKernel))
+            << "Unexpected kernel. Actual: \"" << targetOperationKernel << "\" Expected: \"" << permissibleKernel << "\""
+            << " (Layer type: " << layerType << ", Layer name: " << targetOperationName << ")";
+    }
+}
+
+std::string LayerTestsCommon::getKernel(std::string layerName) {
+    auto perfomanceCounters = inferRequest.GetPerformanceCounts();
+    auto it = perfomanceCounters.find(layerName);
```

Comment:
`ASSERT_NE(perfomanceCounters.end(), it) << " performance counter for node " << layerName << " was not found";` (but not possible if method returns value) or `if (it == perfomanceCounters.end()) { THROW_IE_EXCEPTION << "..." }` (it's your case)